playwright/components/Transactions.ts

```diff
@@ -0,0 +1,173 @@
+import { Page, Locator } from '@playwright/test';
+import {delay} from "../utils/utils";
+
+const SELECTORS = {
+
+    TRANSACTION: {
+        CONTAINER: '.card',
+        VIEW_ALL_TRANSACTIONS: 'a:has-text("View All Transactions")',
+        BASE: '[data-selector="transactions-list"] .tile',
+        TYPE: '[data-test="transaction_type"]',
+        STATUS: '[data-test="transaction_status"]',
+        HASH: '[data-test="transaction_hash_link"]',
+        FROM_ADDRESS: ':nth-child(1) > [data-test="address_hash_link"] > span:first-child',
+        TO_CONTRACT_ADDRESS: '[data-test="address_hash_link"] .contract-address',
+        VALUE: '.tile-title',
+        FEE: '.tile-title + span',
+        BLOCK: 'a[href^="/block/"]',
+        TIMESTAMP: '[data-from-now]'
+    }
+};
+
+export class Transactions {
+    readonly page: Page;
+    readonly container: Locator;
+
+    constructor(page: Page) {
+        this.page = page;
+        this.container = page.locator(SELECTORS.TRANSACTION.CONTAINER);
+    }
+
+    async clickViewAllTransactions() {
+        const viewAllButton = this.container.locator(SELECTORS.TRANSACTION.VIEW_ALL_TRANSACTIONS);
+        await viewAllButton.click();
+    }
+
+    async getAllTransactions(): Promise<TransactionRow[]> {
+        const transactionCount = await this.container.locator(SELECTORS.TRANSACTION.BASE).count();
+        return Array.from({ length: transactionCount }).map((_, i) =>
+            new TransactionRow(this.page)
+        );
+    }
+
+    async findTransactionByHash(hash: string): Promise<TransactionRow | null> {
+        try {
+            const transactionCount = await this.container.locator(SELECTORS.TRANSACTION.BASE).count();
+            for (let i = 1; i <= transactionCount; i++) {
+                const row = new TransactionRow(this.page, hash);
```

Comment:
I don't quite follow what's happening here. All the TransactionRows are being initialized w the same hash

Reply:
Yes this doesnt make sense :) Will sort it.